lib/API/DX/Device.cpp

```diff
@@ -521,51 +528,115 @@ class DXDevice : public offloadtest::Device {
+                   << ", Space = " << R.DXBinding.Space;
+
+      if (R.TilesMapped)
+        llvm::outs() << ", TilesMapped = " << *R.TilesMapped;
+      llvm::outs() << " }\n";
 
       ComPtr<ID3D12Resource> Buffer;
-      if (auto Err = HR::toError(
-              Device->CreateCommittedResource(
-                  &HeapProp, D3D12_HEAP_FLAG_NONE, &ResDesc,
-                  D3D12_RESOURCE_STATE_COMMON, nullptr, IID_PPV_ARGS(&Buffer)),
-              "Failed to create committed resource (buffer)."))
+      if (auto Err =
+              HR::toError(Device->CreateReservedResource(
+                              &ResDesc, D3D12_RESOURCE_STATE_COMMON, nullptr,
+                              IID_PPV_ARGS(&Buffer)),
+                          "Failed to create reserved resource (buffer)."))
         return Err;
 
+      // Committed upload buffer
       ComPtr<ID3D12Resource> UploadBuffer;
       if (auto Err = HR::toError(
               Device->CreateCommittedResource(
-                  &UploadHeapProp, D3D12_HEAP_FLAG_NONE, &UploadResDesc,
+                  &UploadHeapProps, D3D12_HEAP_FLAG_NONE, &UploadResDesc,
                   D3D12_RESOURCE_STATE_GENERIC_READ, nullptr,
                   IID_PPV_ARGS(&UploadBuffer)),
               "Failed to create committed resource (upload buffer)."))
         return Err;
 
-      // Initialize the SRV data
+      // Tile mapping setup (only skipped when TilesMapped is set to 0)
+      const UINT NumTiles = getNumTiles(R.TilesMapped, ResDesc.Width);
+      ComPtr<ID3D12Heap> Heap; // optional, only created if NumTiles > 0
+
+      if (NumTiles > 0) {
+        std::vector<D3D12_TILED_RESOURCE_COORDINATE> StartCoords(NumTiles);
+        std::vector<D3D12_TILE_REGION_SIZE> RegionSizes(NumTiles);
+        std::vector<D3D12_TILE_RANGE_FLAGS> RangeFlags(
+            NumTiles, D3D12_TILE_RANGE_FLAG_NONE);
+        std::vector<UINT> HeapRangeStartOffsets(NumTiles);
+        std::vector<UINT> RangeTileCounts(NumTiles, 1);
+
+        // Create a Heap large enough for the mapped tiles
+        D3D12_HEAP_DESC HeapDesc = {};
+        HeapDesc.Properties = CD3DX12_HEAP_PROPERTIES(D3D12_HEAP_TYPE_DEFAULT);
+        HeapDesc.Alignment = D3D12_DEFAULT_RESOURCE_PLACEMENT_ALIGNMENT;
+        HeapDesc.SizeInBytes = static_cast<UINT64>(NumTiles) *
+                               D3D12_DEFAULT_RESOURCE_PLACEMENT_ALIGNMENT;
+        HeapDesc.Flags = D3D12_HEAP_FLAG_ALLOW_ALL_BUFFERS_AND_TEXTURES;
+
+        if (auto Err =
+                HR::toError(Device->CreateHeap(&HeapDesc, IID_PPV_ARGS(&Heap)),
+                            "Failed to create heap for tiled SRV resource."))
+          return Err;
+
+        // Fill tile coordinates and region sizes
+        for (UINT I = 0; I < NumTiles; ++I) {
+          StartCoords[I] = {I, 0, 0, 0};
+          RegionSizes[I].NumTiles = 1;
+          RegionSizes[I].UseBox = FALSE;
+          HeapRangeStartOffsets[I] = I;
+        }
```

Comment:
There was some talk about changing this to work with a bitmask of tiles to map, or maybe just a list of which tiles to map?  Is this planned for a future change, or are we not going to do it at all?

If we're always going to be mapping a contiguous range of tiles then this can be simplified.

Reply:
Ok, simplified. that bitmask idea should be a future change.